errors/TypeSynonymInstance.md

```diff
@@ -3,19 +3,43 @@
 ## Example
 
 ```purescript
-module ShortFailingExample where
+module TypeSynonymInstanceError where
+import Prelude
 
-...
+type Mass = Number
+
+instance showMass :: Show Mass where
+  show mass = "Mass: " <> show mass <> "kg"
 ```
 
 ## Cause
 
-Explain why a user might see this error.
+*Type synonyms* are merely aliases (usually created to avoid verbose type annotations) and do not declare a new unique type. It is still possible to use the synonym and its equivalent type interchangeably:
+
+```purescript
+velocity = 123.456 :: Mass  -- oops
+```
+
+Since every value that structually matches the type synonym could be considered of the same type, it is not allowed to implement type classes based on type synonyms.
 
 ## Fix
 
-- Suggest possible solutions.
+If a type is wrapped in a **newtype** instead, it becomes a different type than the underlying type. Newtypes can be used as type class instances:
```

Comment:
Used @LiamGoodacre's rewording except for the last sentence. It would be good as an extra example, including code. But I guess if someone gets this error, it is because they tried to declare an explicit type class instance, and are not using `deriving`. Or maybe it belongs into the "Notes" section...?